src/chapters/razor-views/template-forms.rst

```diff
@@ -0,0 +1,114 @@
+Razor Forms
+===========
+
+Already we have seen that templates allow us to build generic forms.
+Using templates, you can reuse the structure by rendering the same form, but with different labels and data.
+Thus, a single form can serve different purposes, saving you extra effort.
+
+Whenever possible, reuse existing templates!
+
+Start a New Project
+-------------------
+
+You will build a new project so you can practice with templates and forms.
+If you have not done so, commit and push any unsaved work from your ``HelloASPDotNET`` project.
+
+.. TODO: Add link back to setup steps
+
+Your new project will keep track of some fictional coding events.
+To get started, follow the steps you took to create ``HelloASPDotNET``, but call the project ``CodingEvents``.
+
+CodingEvents Setup - Video
+--------------------------
+
+.. TODO: Add video
+
+CodingEvents Setup - Text
+-------------------------
+
+#. In the ``Controllers`` directory, create a new controller named ``EventsController``.
+#. In the new controller, create an action method for ``GET`` requests that takes a single parameter. 
+#. Within the action method, create an empty list and add a few event names to it.
+#. Add the list to ``ViewBag``. Then return the corresponding view.
+#. Within the ``Views`` directory, create a new directory named ``Events``. Within this directory, create a new view named ``Index.cshtml``.
+#. Within the new template, loop over the list and display the name of each event.
+
+Create and Render a Form - Video
+--------------------------------
+
+.. TODO: Add video
+
+Create and Render a Form - Text
+-------------------------------
+
+A Razor form can be made simply with a template that includes a ``<form>`` element.
+The method for the form should be of type ``post``.
+
+.. sourcecode:: HTML
+   :linenos:
+
+   <!-- Other HTML -->
+
+   <form method="post">
+      <input type="text" name="inputName">
+      <input type="submit" value="submitButtonText">
+   </form>
+
+   <!-- Other HTML -->
+
+You can include as many inputs as you need in the form, and these can be of
+different types (e.g. text, email, checkbox, etc.). However, each different
+piece of data you want to collect needs to have a unique ``name`` attribute.
+
+To *render* the form in the view, add a method to the controller using the
+``[HttpGet]`` annotation:
+
+.. sourcecode:: csharp
+   :linenos:
```

Comment:
Do we have the finished video/lesson code branch? If so, making the line numbers match up with that code would be good. (Maybe you are already doing this as part of the process...if so, please ignore!)